src/reformatters/common/dynamical_dataset.py

```diff
@@ -43,13 +43,15 @@ class DynamicalDataset(FrozenBaseModel, Generic[DATA_VAR, SOURCE_FILE_COORD]):
     template_config: TemplateConfig[DATA_VAR]
     region_job_class: type[RegionJob[DATA_VAR, SOURCE_FILE_COORD]]
 
-    storage_config: StorageConfig
+    primary_storage_config: StorageConfig
+    replica_storage_configs: list[StorageConfig] = Field(default_factory=list)
```

Comment:
would this work? that way it's not mutable and the specific container type isn't important
```suggestion
    replica_storage_configs: Sequence[StorageConfig] = Field(default_factory=tuple)
```